streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -122,11 +122,12 @@ public final ThreadCache cache() {
 
     /**
      * @throws ProcessorStateException if there is an error while closing the state manager
+     * @param writeCheckpoint boolean indicating if a checkpoint file should be written
      */
-    void closeStateManager() {
+    void closeStateManager(final boolean writeCheckpoint) {
```

Comment:
Could we collapse them into the `ackedOffsets` still where `null` indicates do not write to checkpoint file?